tools/wpt.ts

```diff
@@ -539,7 +539,25 @@ function discoverTestsToRun(
     for (const key in parentFolder) {
       const entry = parentFolder[key];
 
-      if (Array.isArray(entry)) {
+      const expectation = Array.isArray(parentExpectation) ||
```

Comment:
Not entirely happy with the implementation here.

1) in most cases `.html` tests are not runnable in Deno because they require DOM. The ones that are not should be moved to `.any.js` tests. WPT supports this effort, because it means more tests will be run from both window and workers (.html tests are just runnable from window).
2) the current implementation does not handle test filtering.
3) it duplicates code with the else statement below.

Reply:
Okay, I'll send a PR to WPT.

Reply:
Let me know when you open it. I'll cherrypick it into our fork.

Reply:
https://github.com/web-platform-tests/wpt/pull/29014

Reply:
Updated submodule, and reverted the relevant changes in tools/wpt.ts. Thanks :-)